library/std/src/sys/unix/os.rs

```diff
@@ -471,6 +471,43 @@ pub fn current_exe() -> io::Result<PathBuf> {
     if !path.is_absolute() { getcwd().map(|cwd| cwd.join(path)) } else { Ok(path) }
 }
 
+#[cfg(target_os = "aix")]
+pub fn current_exe() -> io::Result<PathBuf> {
+    use crate::io::ErrorKind;
+
+    #[cfg(test)]
+    use realstd::env;
+
+    #[cfg(not(test))]
+    use crate::env;
+
+    let exe_path = env::args().next().ok_or(io::const_io_error!(
+        ErrorKind::Uncategorized,
+        "an executable path was not found because no arguments were provided through argv"
+    ))?;
+    let path = PathBuf::from(exe_path);
+    if path.is_absolute() {
+        return path.canonicalize();
+    }
+    // Search PWD to infer current_exe.
+    if let Some(pstr) = path.to_str() && pstr.contains("/") {
+        return getcwd().map(|cwd| cwd.join(path))?.canonicalize();
+    }
+    // Search PATH to infer current_exe.
```

Comment:
Are we searching PATH and PWD on other platforms today? I'm not sure we should do that here -- it increases the amount of work done and if it's not done on other platforms is somewhat inconsistent (and could have false positives much more easily).

Reply:
Right, but you're using argv[0] just above which is what we do on other platforms (AFAICT, looking in this file, for example).

I am inclined to leave it there unless there's strong motivation for doing this cwd/path traversal. current_exe is documented as fallible for a reason.

Reply:
Apologies if I wasn't clear. I think we should *remove* the PATH searching from this function, just returning the passed argument. IOW, I think if just adding aix to the fuchsia function works, then that seems like the right initial implementation here.